Add support for chef-vault to data_bag_item helper

Afterglow opened the pull request:
Attempts to decrypt an encrypted data bag with chef_vault_item if the corresponding `item_keys.json` exists otherwise continues to use standard encryption.

Tested this with the encrypted-ubuntu-1404 and vault-ubuntu-1404. I had a bunch of other problems that look unrelated trying to get centos or 1204 test kitchens to converge. Would appreciate input from anyone else on those as i'm not sure if its my setup or what.

    ~/w/i/sensu-chef *vault> bundle exec kitchen verify vault-ubuntu-1404
    -----> Starting Kitchen (v1.2.1)
    -----> Verifying <vault-ubuntu-1404>...
           Removing /tmp/busser/suites/serverspec
    Uploading /tmp/busser/suites/serverspec/dependency_spec.rb (mode=0644)
    Uploading /tmp/busser/suites/serverspec/encrypted_spec.rb (mode=0644)
    -----> Running serverspec test suite
    /opt/chef/embedded/bin/ruby -I/tmp/busser/suites/serverspec -I/tmp/busser/gems/gems/rspec-support-3.1.2/lib:/tmp/busser/gems/gems/rspec-core-3.1.7/lib /opt/chef/embedded/bin/rspec --pattern /tmp/busser/suites/serverspec/\*\*/\*_spec.rb --color --format documentation --default-path /tmp/busser/suites/serverspec

    File "/opt/sensu/embedded/bin/ruby"
      should be file
      should be executable

    Command "ps aux | grep rabbitmq-server | grep -v grep"
      exit_status
        should eq 0

    Port "5671"
      should be listening
    
    Command "ps aux | grep redis-server | grep -v grep"
      exit_status
        should eq 0
    
    Port "6379"
      should be listening
    
    Service "sensu-server"
      should be enabled
      should be running
    
    Service "sensu-client"
      should be enabled
      should be running
    
    Service "sensu-api"
      should be enabled
      should be running
    
    File "/etc/sensu/config.json"
      content
        should match /vaultpassword/
    
    Finished in 0.30233 seconds (files took 0.32515 seconds to load)
    13 examples, 0 failures
           Finished verifying <vault-ubuntu-1404> (0m2.69s).
    -----> Kitchen is finished. (0m3.11s)

portertech commented on libraries/sensu_helpers.rb:
```diff
@@ -43,8 +43,12 @@ def data_bag_item(item, missing_ok=false)
         if encrypted
-          secret = Chef::EncryptedDataBagItem.load_secret
-          Chef::EncryptedDataBagItem.new(raw_hash, secret)
+          if Chef::DataBag.load("sensu").key? "#{item}_keys"
```
Perhaps use `has_key?()` here?

Afterglow commented:
I've updated it to use `key?()` and also the one other instance of `has_key?` in the cookbook i've changed to `key?` as well further up in the file. Ruby devs indicate that `has_key?` is deprecated in favour of `key?`

Afterglow commented:
This converges properly in 12.04 with the fixes to 12.04 rabbitmq from #312 

    vault-ubuntu-1204        Vagrant  ChefZero         Verified

Afterglow commented:
messed up the history will recreate this PR off a new branch
